notebook/notebookapp.py

```diff
@@ -1566,15 +1564,23 @@ def init_server_extensions(self):
         manager = ConfigManager(read_config_path=config_path)
         section = manager.get(self.config_file_name)
         extensions = section.get('NotebookApp', {}).get('nbserver_extensions', {})
+        
+        for modulename, enabled in sorted(extensions.items()):
+            if modulename not in self.nbserver_extensions:
+                self.config.NotebookApp.nbserver_extensions.update({modulename: enabled})
+                self.nbserver_extensions.update({modulename: enabled})
```

Comment:
👍

Reply:
👍 This is the most important change, I think. `self.nbserver_extensions` is the thing to check, and that wasn't properly updated. It should never be assumed that `self.config.NotebookApp.nbserver_extensions` contains the final value of the extensions to be loaded.